### PR TITLE
Update CI to use ubuntu18.04

### PR DESCRIPTION
**Pull request by JunTaoLuo**
WIP debugging systemd failure, not ready for review.

### CODE REVIEW

File: src/Hosting/Systemd/test/Microsoft.Extensions.Hosting.Systemd.Tests.csproj
Comment (JunTaoLuo) [inline src/Hosting/Systemd/test/Microsoft.Extensions.Hosting.Systemd.Tests.csproj:19]: This was added to add a reference so on test failure a more reasonable assert failure will occur instead of a cryptic assembly load error.

File: src/Hosting/Systemd/test/UseSystemdTests.cs
Comment (JunTaoLuo) [inline src/Hosting/Systemd/test/UseSystemdTests.cs:53]: @Tratcher Looks like this check isn't functioning properly on ubuntu 18.04. From the test logs:
```
[m[31;1m    Microsoft.Extensions.Hosting.UseSystemdTests.DefaultsToOffOutsideOfService [FAIL]
[m[37m      Assert.IsType() Failure
[m[37m      Expected: Microsoft.Extensions.Hosting.Internal.ConsoleLifetime
[m[37m      Actual:   Microsoft.Extensions.Hosting.Systemd.SystemdLifetime
[m[30;1m      Stack Trace:
[m[37m        /_/src/Hosting/Systemd/test/UseSystemdTests.cs(38,0): at Microsoft.Extensions.Hosting.UseSystemdTests.DefaultsToOffOutsideOfService()
[m[30;1m      Output:
[m[37m        Invocation ID obtained: 2303132d69eb43498bc4ce6a0556d9ae
```

Looks like invocation_id is not empty even though we didn't launch via systemd. I'm not sure how updating to ubuntu18.04 can cause this nor did a quick search yield any insights. Any ideas on how to investigate further?

**Comment by JunTaoLuo** [comment]
Note, attempting to port https://github.com/dotnet/extensions/commit/cda774390962745f07baa11069f46aec489f4c61#diff-a7308f4fc9f48527c77a60e5a21d73a56887f15f8f965fb81d01012696f3449a to see if stricter checks works on ubuntu 18.04. 

DO NOT MERGE THIS CHANGE since it's a behaviour change. If this works, we'll just disable the test. We won't backport the actual change unless it meets the servicing bar.

**Comment by JunTaoLuo** [comment]
Looks like the stricter checks currently in dotnet/runtime main would fix the tests on ubuntu 18.04. I'll revert the changes and disable the test in this repo for release/3.1 to unblock the ubuntu18.04 update. We won't actually backport the functional change unless there's a servicing request to do so.

**Review by dougbu — APPROVED** [review]

**Comment by wtgodbe** [comment]
@JunTaoLuo branches are closed

**Comment by JunTaoLuo** [comment]
Oops, I meant to merge this once the branches were opened for the next release, do I need to revert?

**Comment by wtgodbe** [comment]
> Oops, I meant to merge this once the branches were opened for the next release, do I need to revert?

No worries, I cancelled the internal PRs before they got too far. Don't revert, that would just trigger another build/more dependency flow

**Pull request merged**
